### DISCUSSION TITLE
How to record Z value automatically

### DISCUSSION THREAD

**Discussion by xyzadh** [opengisch/QField#3580] [Q&A]
I was wondering how to record Z value when I'm mapping out points. I use Qfield and Emlid reach. An answer on GIS stack exchange says to put Z($geometry). I was wondering, does just $Z work as well? What does the Z($geometry) do differently? And, is it possible to use a GNSS variable for Z value. I would rather want the field to be empty if there is a sudden loss of connection to base. 

And, Qfield does not seem to correct the altitude height. Any ideas?

**Comment by kstulka** [comment]
Hi

I noticed, you have to create point later with Z field. When you add geoid file *.tiff to qfield then you can save the height value with either $z or Z($geometry).
